packages/connect-form/src/components/advanced-options-tabs/advanced-tab/url-options-modal.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { ChangeEvent, KeyboardEvent } from 'react';
+import { css } from '@emotion/css';
+import {
+  spacing,
+  uiColors,
+  Button,
+  Modal,
+  H3,
+  TextInput,
+  FormFooter,
+  Select,
+  Option,
+  OptionGroup,
+} from '@mongodb-js/compass-components';
+
+import FormFieldContainer from '../../form-field-container';
+import { editableUrlOptions, UrlOption } from '../../../utils/url-options';
+
+const modalContainerStyles = css({
+  padding: 0,
+  'button[aria-label="Close modal"]': {
+    position: 'absolute',
+  },
+});
+
+const modalContentStyles = css({
+  padding: spacing[5],
+});
+
+// copied from collection-fields/time-series-fields.module.less
+const selectStyles = css({
+  zIndex: 1,
+  'button:focus': {
+    zIndex: 20,
+  },
+  'button:focus-within': {
+    zIndex: 20,
+  },
+});
+
+const modalFooterStyles = css({
+  boxShadow: 'none',
+  border: 'none',
+  borderTop: `1px solid ${uiColors.gray.light2}`,
+});
+
+function UrlOptionsModal({
+  onClose,
+  selectedOption,
+  onUpdateOption,
+}: {
+  onClose: (isOpen: boolean) => void;
+  onUpdateOption: (option: UrlOption) => void;
+  selectedOption?: UrlOption;
+}): React.ReactElement {
+  const [errorMessage, setErrorMessage] = React.useState('');
+  const [option, setOption] = React.useState(
+    selectedOption ?? { name: '', value: '' }
+  );
+
+  const handleKeyPress = (event: KeyboardEvent<HTMLInputElement>): void => {
+    if (event.key === 'Enter') {
+      event.preventDefault();
+      addUrlOption();
+    }
+  };
+
+  const addUrlOption = () => {
+    setErrorMessage('');
+    if (!option.name) {
+      return setErrorMessage('Please select an options key.');
+    }
+    onUpdateOption(option as UrlOption);
+  };
+
+  return (
+    <Modal
+      contentClassName={modalContainerStyles}
+      open={true}
+      setOpen={onClose}
+      data-testid="uri-options-modal"
+    >
+      <div className={modalContentStyles}>
+        <H3>Add custom url option</H3>
+        <FormFieldContainer>
+          <Select
```

Comment:
This select seems to be working for me when I'm using keyboard (with slight visual glitches on select), but not at all when I'm using the mouse. This is probably not expected, right? 🤔

Reply:
removed modal completely. I think the glitch was due to the `event.preventDefault` not being called.